harmonica/forward/utils.py

```diff
@@ -178,3 +187,130 @@ def distance_spherical_core(
     cospsi = sinphi_p * sinphi + cosphi_p * cosphi * coslambda
     dist = np.sqrt((radius - radius_p) ** 2 + 2 * radius * radius_p * (1 - cospsi))
     return dist, cospsi, coslambda
+
+
+def distance_geodetic(point_p, point_q, ellipsoid):
+    """
+    Calculate the distance between two points in geodetic coordinates
+
+    Computes the Euclidean distance between two points given in geodetic
+    coordinates using the closed-form formula given by [Vajda2004]_.
+    All angles must be in degrees and height above the ellipsoid in meters.
+
+    Parameters
+    ----------
+    point_p : tuple or 1d-array
+        Tuple or array containing the coordinates of the first point in the
+        following order: (``longitude``, ``latitude`` and ``height``).
+        Both ``longitude`` and ``latitude`` must be in degrees, while
+        ``height`` in meters.
+    point_q : tuple or 1d-array
+        Tuple or array containing the coordinates of the second point in the
+        following order: (``longitude``, ``latitude`` and ``height``).
+        Both ``longitude`` and ``latitude`` must be in degrees, while
+        ``height`` in meters.
+    ellipsoid : :class:`boule.Ellipsoid`
+        Reference ellipsoid for the geodetic coordinates of points ``point_p``
+        and ``point_q``. Must be a instance of :class:`boule.Ellipsoid`.
+
+    Returns
+    -------
+    distance : float
+        Euclidean distance between ``point_p`` and ``point_q``.
+
+    Example
+    -------
+
+    >>> import boule as bl
+    >>> ellipsoid = bl.WGS84
+    >>> point_p = (-72.3, -33.3, 644)
+    >>> point_q = (-70.1, -31.6, 1024)
+    >>> distance_geodetic(point_p, point_q, ellipsoid)
```

Comment:
```suggestion
    >>> point_p = (-72.3, -33.3, 644)
    >>> point_q = (-70.1, -31.6, 1024)
    >>> distance = distance_geodetic(point_p, point_q, ellipsoid)
    >>> print("{:.2f} m".format(distance))
    279878.84 m
```

Reply:
I was waiting for Boule 0.2 to be out in order to get the expected example result and to run the tests to see if the closed-form formula works fine.